### DISCUSSION TITLE
Strange classes generated at the end of output.css

### DISCUSSION THREAD

**Discussion by vitality82** [tailwindlabs/tailwindcss#8868] [Help] [answered]
Hi, 

I have successfully integrated the standalone CLI with an older project. I'm curious what is the purpose of the following classes generated at the end of the output.css

```
.\[slug\:board\.slug\] {
  slug: board.slug
}

.\[board\:board\] {
  board: board
}
```

Much appreciated.

**Comment by adamwathan** [comment, marked as answer]
Hey! So this is because Tailwind sees the strings `[slug:board.slug]` and `[board:board]` somewhere in your project and thinks you are trying to use [arbitrary properties](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties).

I think in a perfect world we'd find a way to not generate these when the property isn't a real CSS property, but not sure the best way to do it off the top of my head 🤔

**Comment by vitality82** [comment]
Found it, so the following is a snippet from a Grails template (filename: _menu.gsp):

`<g:link controller="public" action="year" params="${[slug:board.slug]}">Meetings</g:link>`

Tailwind prob evaluates the file prior to grails resolving the g.link tag into an HTML link, and sees the slug:board.slug as an arbitrary prop? What if the tailwind watcher was updated with an attribute ignore list? 

Thanks!

**Reply by adamwathan** [reply]
So Tailwind doesn't actually try to understand your content at all, so an attribute ignore list wouldn't really work unfortunately.

We have to handle cases like:

```
let someVar = "bg-red-500 [mask-type:luminance]"

document.getElementById('foo').classList.add(someVar)
```

...so we literally just regex against the entire file without making any attempt to understand it at all, otherwise we'd need special parsers for every language which I don't think would be sustainable.

I'm going to look into at least adding a `blocklist` feature so people can prevent this sort of thing from generating unnecessary CSS if it's important to them.

**Reply by vitality82** [reply]
That makes sense! 
I do like your 'blocklist' idea and might actually be useful to a wider audience as well.

Thanks for your help ✌️
